cloud/gcp/pubsub/subscription/variables.tf

```diff
@@ -0,0 +1,279 @@
+  description = "Notification recipients list for critical alerting rule of oldest_unacked_message detector"
+  type        = list
+  default     = []
+}
+
+variable "oldest_unacked_message_aggregation_function" {
+  description = "Aggregation function and group by for oldest_unacked_message detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ""
+}
+
+variable "oldest_unacked_message_transformation_function" {
+  description = "Transformation function for oldest_unacked_message detector (mean, min, max)"
+  type        = string
+  default     = "min"
+}
+
+variable "oldest_unacked_message_transformation_window" {
+  description = "Transformation window for oldest_unacked_message detector (i.e. 5m, 20m, 1h, 1d)"
+  type        = string
+  default     = "5m"
+}
+
+variable "oldest_unacked_message_threshold_critical" {
+  description = "Critical threshold for oldest_unacked_message detector"
+  type        = number
+  default     = 120
+}
+
+variable "oldest_unacked_message_threshold_warning" {
+  description = "Warning threshold for oldest_unacked_message detector"
+  type        = number
+  default     = 30
+}
+
+# Push_latency detectors
+
+variable "push_latency_disabled" {
+  description = "Disable all alerting rules for push_latency detector"
+  type        = bool
+  default     = null
+}
+
+variable "push_latency_disabled_critical" {
+  description = "Disable critical alerting rule for push_latency detector"
+  type        = bool
+  default     = null
+}
+
+variable "push_latency_disabled_warning" {
+  description = "Disable warning alerting rule for push_latency detector"
+  type        = bool
+  default     = null
+}
+
+variable "push_latency_notifications" {
+  description = "Notification recipients list for every alerting rules of push_latency detector"
+  type        = list
+  default     = []
+}
+
+variable "push_latency_notifications_warning" {
+  description = "Notification recipients list for warning alerting rule of push_latency detector"
+  type        = list
+  default     = []
+}
+
+variable "push_latency_notifications_critical" {
+  description = "Notification recipients list for critical alerting rule of push_latency detector"
+  type        = list
+  default     = []
+}
+
+variable "push_latency_aggregation_function" {
+  description = "Aggregation function and group by for push_latency detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ""
+}
+
+variable "push_latency_transformation_function" {
+  description = "Transformation function for push_latency detector (mean, min, max)"
+  type        = string
+  default     = "mean"
+}
+
+variable "push_latency_transformation_window" {
+  description = "Transformation window for push_latency detector (i.e. 5m, 20m, 1h, 1d)"
+  type        = string
+  default     = "10m"
+}
+
+variable "push_latency_threshold_critical" {
+  description = "Critical threshold for push_latency detector"
```

Comment:
Metric unit for `subscription/push_request_latencies` on [google api](https://cloud.google.com/monitoring/api/metrics_gcp#gcp-pubsub) is "microseconds".
Shouldn't be the threshold 5000000 instead of 5000?

Same for warning threshold.

Reply:
good catch ! the error was also in the original implementation : https://github.com/claranet/terraform-datadog-monitors/blob/master/cloud/gcp/pubsub/subscription/inputs.tf#L120